ci/acceptance-tests.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+
+set -e
+set -u
+
+# Set defaults
+TTL="${TTL:-60}"
+
+suffix="${RANDOM}"
+DOMAIN=$(printf "${DOMAIN}" "${suffix}")
+SERVICE_INSTANCE_NAME=$(printf "${SERVICE_INSTANCE_NAME}" "${suffix}")
+
+# Authenticate
+cf api "${CF_API_URL}"
+cf auth "${CF_USERNAME}" "${CF_PASSWORD}"
+
+# Target
+cf target -o "${CF_ORGANIZATION}" -s "${CF_SPACE}"
+
+# Create service
+opts=$(jq -n --arg domain "${DOMAIN}" --arg origin "${ORIGIN}" '{domain: $domain, origin: $origin}')
+cf create-service "${SERVICE_NAME}" "${PLAN_NAME}" "${SERVICE_INSTANCE_NAME}" -c "${opts}"
+
+# Get CNAME instructions
+regex="CNAME domain (.*) to (.*)\.$"
+
+elapsed=60
+until [ "${elapsed}" -le 0 ]; do
+  message=$(cf service "${SERVICE_INSTANCE_NAME}" | grep "^Message: ")
+  if [[ "${message}" =~ ${regex} ]]; then
+    external="${BASH_REMATCH[1]}"
+    internal="${BASH_REMATCH[2]}"
+    break
+  fi
+  let elapsed-=5
+  sleep 5
+done
+if [ -z "${internal}" ]; then
+  echo "Failed to parse message: ${message}"
+  exit 1
+fi
+
+# Create CNAME record
+cat << EOF > ./create-cname.json
+{
+  "Changes": [
+    {
+      "Action": "CREATE",
+      "ResourceRecordSet": {
+        "Name": "${external}.",
+        "Type": "CNAME",
+        "TTL": ${TTL},
+        "ResourceRecords": [
+          {"Value": "${internal}"}
+        ]
+      }
+    }
+  ]
+}
+EOF
+aws route53 change-resource-record-sets \
+  --hosted-zone-id "${HOSTED_ZONE_ID}" \
+  --change-batch file://./create-cname.json
+
+# Wait for provision to complete
+elapsed=3600
+until [ "${elapsed}" -le 0 ]; do
+  status=$(cf service "${SERVICE_INSTANCE_NAME}" | grep "^Status: ")
+  if [[ "${status}" =~ "Status: create succeeded" ]]; then
+    updated="true"
+    break
+  elif [[ "${status}" =~ "Status: create failed" ]]; then
```

Comment:
Maybe just check for "succeeded"/"failed"? I worry that this could be brittle if the language/punctuation changes.